### PR TITLE
Local infra

### PR DESCRIPTION
**Pull request by mmamayka**
Adds:
- clang-format
- CMake stub + test stub
- google test submodule dependency
- entry point stub main.cpp
- YCM (vim syntax highlight plugin) environment

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+BUILD_DIR := build
+SRC_DIR := src
+
+JOBS ?= 4
+
+.PHONY: all
+all: build test
```
Comment (c71n93) [inline Makefile:8]: I think we need add `format` to this target.

File: .ycm_extra_conf.py
Comment (c71n93) [inline .ycm_extra_conf.py:1]: I think we need to remove this file))

**Review by levBagryansky — DISMISSED** [review]
@mmamayka LGTM 👍👍👍👍👍

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+.PHONY: format
+.SILENT: format
+format:
+	./tools/clang-format.sh $(SRC_DIR)
+
+.PHONY: build
+build:
```
Comment (levBagryansky) [inline Makefile:16]: @mmamayka If we need `git submodule init` command to be executed before, maybe it is better call it here too?

File: Makefile (hunk 1 above)
Comment (mmamayka) [reply Makefile]: Done

File: Makefile (hunk 2 above)
Comment (mmamayka) [reply Makefile]: Done

**Review by c71n93 — APPROVED** [review]
Okaaaay let's go

**Pull request merged**